src/frog-base.tsx

```diff
@@ -256,6 +251,13 @@ export class FrogBase<
       } = await handler(context)
       const buttonValues = getButtonValues(parseIntents(intents))
 
+      if (context.status === 'redirect' && context.buttonIndex) {
+        const buttonValue = buttonValues[context.buttonIndex - 1]
+        const location = buttonValue?.replace(/^_r:/, '')
+        if (!location) throw new Error('location required to redirect')
+        return c.redirect(location, 302)
+      }
```

Comment:
Needed to move this down so we have resolved `buttonValues` based on the `context` from the current action. Otherwise, redirect `location` is based on old context values.